lib/utils.js

```diff
@@ -378,11 +378,12 @@ const executeOperation = (topology, operation, args, options) => {
 
   // The driver sessions spec mandates that we implicitly create sessions for operations
   // that are not explicitly provided with a session.
-  let session, opOptions;
+  let session, opOptions, owner;
   if (!options.skipSessions && topology.hasSessionSupport()) {
     opOptions = args[args.length - 2];
     if (opOptions == null || opOptions.session == null) {
-      session = topology.startSession();
+      owner = {};
```

Comment:
can't this just be left undefined/null?

Reply:
No, it needs to be a reference that is owned by the "context".

Reply:
as in you need a reference in order to prevent garbage collection?

Reply:
oh, you just need it for comparison below? Can't you just use `this`, being a reference to the function?

Reply:
`this` won't be a reference to the function, but rather the global scope.

```
console.log(this === global); //
function foo(x) {
  console.log(x === this);
}
foo(global); //true
```

Reply:
I guess my point is that we _should_ be able to find a way to do this without an allocation. This is a method that's called for _every_ piece of public API, and the implicit sessions path is even more likely to be run. Can we maybe use the passed in `operation` as a potential owner? Seems like you really just need something to uniquely identify the callsite, in order to avoid cleaning up the session before it's truly done.